compiler/dyno/include/chpl/util/subprocess.h

```diff
@@ -34,19 +34,25 @@ namespace chpl {
  commandVec
    first element is the command to run, the rest are the arguments.
 
+ envVec
+   a list of environment variables represented as strings
+
  description
    a string representation of the command to run
 
  printSystemCommands if true, print the command to be run
 
  returns
+   a pair of exit code and the program output. the exit code is
```

Comment:
We should just have a different function that saves the program output. E.g. you could port over https://github.com/chapel-lang/chapel/blob/658369b815b200b3624f5fbea09b9200675e72fa/compiler/util/files.cpp#L518-L541 to subprocess.cpp . (I don't think the name `runCommand` is particularly good; we should have a name that indicates it captures the output).

Reply:
This looks good, except for one hitch: the use of `USR_FATAL`. I assume we don't want dyno (i.e., library code) to crash and burn; I can make this work using a return of `std::pair<bool, string>` or something, but it gets somewhat cumbersome to handle this if I wanted to port `runPrintChplEnv` too.

Reply:
Yes it can return an integer error code, or a bool, or even an ErrorMessage. You can return a pair or return the value through an argument. IIRC I had some discussion about the pattern LLVM uses for such things... maybe that was in discussion around PR #20280 and

>  `std::error_code makeTempDir(std::string dirPrefix, std::string& tmpDirPathOut);`

@arezaii - do you recall such a discussion? Since apparently I can't remember it... maybe we can put it in compiler/dyno/README as "style guide" sort of information.

I can see that LLVM's FileSystem.h uses the pattern of returning any error in `std::error_code` and another result (often a string) through an argument.

I can also see LLVM's Program.h uses `ErrorOr<std::string>` in some places.

https://llvm.org/docs/ProgrammersManual.html#error-handling has some guidance on this. My read of the situation is:
 * using `ErrorOr<std::string>` seems appropriate for this function
 * using `std::pair<bool, string>` seems counter to LLVM style
 * it would also be ok to return `Expected<std::string>` or to return a `std::error_code` and set the result string through an argument.

A future work item for us is to make ErrorMessage into a subclass of LLVM's ErrorInfo, so that we can use ErrorMessage with LLVM's `Expected`.